cp3pt0-deployment/common/migrate_singleton.sh

```diff
@@ -81,6 +76,88 @@ function main() {
+    if [[ lsr_instances -eq 0 ]]; then
+        info "No LSR for migration found in ibm-common-services namespace"
+        return 0
+    fi
+    # Prepare LSR PV/PVC which was decoupled in isolate.sh
+    # delete old LSR CR - PV will stay as during isolate.sh the policy was set to Retain
+    ${OC} delete IBMLicenseServiceReporter instance -n ibm-common-services
+
+    # in case PVC is blocked with deletion, the finalizer needs to be removed
+    lsr_pvcs=$("${OC}" get pvc license-service-reporter-pvc -n ibm-common-services  --no-headers | wc -l)
+    if [[ lsr_pvcs -gt 0 ]]; then
+        info "Failed to delete pvc license-service-reporter-pvc, patching its finalizer to null..."
+        ${OC} patch pvc license-service-reporter-pvc -n ibm-common-services  --type="json" -p '[{"op": "remove", "path":"/metadata/finalizers"}]'
+    else
+        debug1 "No pvc license-service-reporter-pvc as expected"
+    fi
+
+    lsr_pv_nr=$("${OC}" get pv -l license-service-reporter-pv=true --no-headers | wc -l )
+    if [[ lsr_pv_nr -gt 1 ]]; then
+        error "More than on PV with label license-service-reporter-pv=true was found. Only one is allowed."
+    fi
+
+    if [[ lsr_pv_nr -eq 1 ]]; then
+        debug1 "LSR namespace: ${LSR_NAMESPACE}" 
+        create_namespace "${LSR_NAMESPACE}"
+
+        # get storage class name
+        LSR_PV_NAME=$("${OC}" get pv -l license-service-reporter-pv=true -o=jsonpath='{.items[0].metadata.name}')
+        debug1 "PV name: $LSR_PV_NAME"
+        
+        # on ROKS storage class name cannot be proviced during PVC creation
+        roks=$(${OC} cluster-info | grep 'containers.cloud.ibm.com')
+        if [[ -z $roks ]]; then
+            LSR_STORAGE_CLASS=$("${OC}" get pv -l license-service-reporter-pv=true -o=jsonpath='{.items[0].spec.storageClassName}')
+            if [[ -z $LSR_STORAGE_CLASS ]]; then
+                error "Cannnot get storage class name from PVC license-service-reporter-pv in $LSR_NAMESPACE"
+            fi
+        else
+            debug1 "Run on ROKS, not setting storageclass name"
+            LSR_STORAGE_CLASS=""
+        fi
+
+        # create PVC
+        TEMP_LSR_PVC_FILE="_TEMP_LSR_PVC_FILE.yaml"
+
+        cat <<EOF >$TEMP_LSR_PVC_FILE
+        apiVersion: v1
+        kind: PersistentVolumeClaim
+        metadata:
+            name: license-service-reporter-pvc
+            namespace: ${LSR_NAMESPACE}
+        spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+                requests:
+                    storage: 1Gi
+            storageClassName: "${LSR_STORAGE_CLASS}"
+            volumeMode: Filesystem
+            volumeName: ${LSR_PV_NAME}
+EOF
+
+        ${OC} create -f ${TEMP_LSR_PVC_FILE}
+        # checking status of PVC - in case it cannot be boud, the claimRef needs to be set to null
+        status=$("${OC}" get pvc license-service-reporter-pvc -n $LSR_NAMESPACE --no-headers | awk '{print $2}')
+        while [[ "$status" != "Bound" ]]
```

Comment:
Do we really need to wait for the PVC binding, given we don't depend on it at all here? For mongo, we had to do the wait time, since we load data into it, but here we don't really need to do so, right?

Reply:
From our tests it seems we need to wait. @martabereta could you comment on details?

Reply:
Here are the logs from the tests - it looks that the loop was in use:
```
# Checking whether Namespace lsr-ns exist...
[INFO] Creating namespace lsr-ns
namespace/lsr-ns created
[INFO] Waiting for project lsr-ns to be created
[✔] Project lsr-ns is created

persistentvolumeclaim/license-service-reporter-pvc created
persistentvolume/pvc-f5b2f55f-0931-41d3-b60c-c2ae05649d66 patched
[INFO] Waiting for pvc license-service-reporter-pvc to bind
persistentvolume/pvc-f5b2f55f-0931-41d3-b60c-c2ae05649d66 patched
[INFO] Waiting for pvc license-service-reporter-pvc to bind
persistentvolume/pvc-f5b2f55f-0931-41d3-b60c-c2ae05649d66 patched
[INFO] Waiting for pvc license-service-reporter-pvc to bind
persistentvolume/pvc-f5b2f55f-0931-41d3-b60c-c2ae05649d66 patched
[INFO] Waiting for pvc license-service-reporter-pvc to bind
persistentvolume/pvc-f5b2f55f-0931-41d3-b60c-c2ae05649d66 patched
[INFO] Waiting for pvc license-service-reporter-pvc to bind
persistentvolume/pvc-f5b2f55f-0931-41d3-b60c-c2ae05649d66 patched (no change)
[INFO] Waiting for pvc license-service-reporter-pvc to bind
[INFO] Waiting for pvc license-service-reporter-pvc to bind
configmap/ibmlicensing-instance-bak created
ibmlicensing.operator.ibm.com "instance" deleted
# Deleting ibm-licensing-operator in namesapce ibm-common-services...
[1] Removing the subscription of ibm-licensing-operator in namesapce ibm-common-services ...
[2] Removing the csv of ibm-licensing-operator in namesapce ibm-common-services ...```

Reply:
I am not asking what the code does. I am asking why do we need to wait at all, if in our script, we don't make any use of the created PVC? It should be up to Reporter operator to deal with that, right?

Reply:
The loop contains not only checking if PVC was bound but in case it is not it bound, it set `claimRef` on PV to null